source/agora/test/Base.d

```diff
@@ -135,28 +135,21 @@ private UnitTestResult customModuleUnitTester ()
         }
         catch (Throwable ex)
         {
-            writefln("Module tests failed: %s", mod.name);
-            writeln(ex);
+            synchronized
```

Comment:
Can we use an explicit mutex here ?

Reply:
I guess so. Is there any reason why we should avoid using the one that's autogenerated for a code block?

Reply:
Explicit lifetime control ?
I have an innate doubt towards any D-generated magic. If you feel strongly about it, I can drop it. But it's one of those things I wish would go away.

Reply:
No it's alright to be explicit about it. I've updated the PR.

I've also added another commit (hopefully not controversial).

Reply:
Actually I remember now I had bugs in my code where I assumed multiple `synchronized {}` blocks used the same lock, but in fact each statement uses its own lock. So you're right, we should be explicit.